src/plugins/dashboard/public/dashboard_container/embeddable/create/create_dashboard.ts

```diff
@@ -424,5 +460,49 @@ export const initializeDashboard = async ({
     setTimeout(() => dashboard.dispatch.setAnimatePanelTransforms(true), 500)
   );
 
-  return { input: initialInput, searchSessionId: initialSearchSessionId };
+  // --------------------------------------------------------------------------------------
+  // Start diffing subscription to keep track of unsaved changes
+  // --------------------------------------------------------------------------------------
+  untilDashboardReady().then((dashboard) => {
+    // subscription that handles the unsaved changes badge
+    dashboard.integrationSubscriptions.add(
```

Comment:
Refer to https://github.com/elastic/kibana/pull/175146/files#r1470029064 for why I have two different subscriptions here....